### PR TITLE
Add RCT Support

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
Adds support for RCT while maintaining backward compatibility with MK2. Re-structures the repository so that other boards can be added later (if we want)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,57 +62,62 @@ OPENOCD := openocd
...
-
 Q := @
 # Do not print "Entering directory ...".
 MAKEFLAGS += --no-print-directory
 endif
 
-# common objects
-OBJS += $(CPU_OBJS) $(BOARD_OBJS) $(APP_OBJS)
+# Object files
+APP_O_FILES = $(APP_C_FILES:.c=.o) $(APP_S_FILES:.s=.o)
```
Comment (stieg) [inline Makefile]: Should this be := or do we want this expanded later?

Comment (Jeff-Ciesielski) [inline Makefile]: Good question.  There is a little bit of meta-make going on in this file (the libraries), so I had been wanting to expand as late as possible.  I can change to an expand-at-assignment and re-structure a little bit to only require this with the libraries.

**Comment by brentpicasso** [comment]
sounds like we should call the project - ASL_STM_bootloader now! :)

File: board/rct/led.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	GPIO_DeInit(GPIOA);
+
+}
+
+void led_toggle(void)
+{
+	const uint8_t led_state = GPIO_ReadInputDataBit(GPIOA, GPIO_Pin_1);
+
+	if (led_state) {
+		GPIO_ResetBits(GPIOA, GPIO_Pin_1);
```
Comment (brentpicasso) [inline board/rct/led.c]: we need to invert this since we've re-configured the hardware so LED on == uC active low (LED is tied to V+)

**Comment by brentpicasso** [comment]
Looks yuuge!

Let's wire up a check for PC15 in bootmode_upgrade_requested() for RCT, and the note about the LED mode, to reflect the production hardware.

**Pull request merged**